data/remote/src/main/java/com/brunotiba/remote/api/WeatherService.kt

```diff
@@ -0,0 +1,53 @@
+package com.brunotiba.remote.api
+
+import com.brunotiba.remote.BuildConfig
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Singleton
+
+/**
+ * Service responsible for fetching data from the weather api.
+ */
+@Singleton
+internal class WeatherService {
+
+    private val weatherApi: WeatherApi
+
+    init {
```

Comment:
Suggestion: instead of property initialization on constructor, you could call it directly on declaration. Usually in Kotlin the constructor is used for other stuff.

Reply:
Done